tracer/test/Datadog.Trace.Tests/Propagators/W3CBaggagePropagatorTests.cs

```diff
@@ -76,16 +75,12 @@ static W3CBaggagePropagatorTests()
     [InlineData("abcd", new[] { 'x' }, "abcd")]                                      // encode a char that is not in the string
     [InlineData("abcd", new[] { 'b', 'd' }, "a%62c%64")]                             // encode chars that are in the string
     [InlineData("José 🐶\t我", new char[0], "Jos%C3%A9%20%F0%9F%90%B6%09%E6%88%91")] // always encode whitespace and unicode chars
-    public void Encode(string value, char[] charsToEncode, string expected)
+    public void EncodeAndAppend(string value, char[] charsToEncode, string expected)
     {
-        var result = W3CBaggagePropagator.Encode(value, [..charsToEncode]);
-        result.Should().Be(expected);
+        var sb = new StringBuilder();
+        W3CBaggagePropagator.EncodeStringAndAppend(sb, value, [..charsToEncode]);
 
-        if (value == expected)
-        {
-            // ensure that the method returns the same string instance when no encoding is needed
-            ReferenceEquals(value, result).Should().BeTrue();
```

Comment:
I guess this is no longer the case - should it be?

Reply:
The test is now always creating a new string via `StringBuilder.ToString()`. Internally, if the string we're adding doesn't need encoding, we append it directly into the `StringBuilder`, which is the "spiritual equivalent" of what we had before (returning the same string instance).